src/Workspaces/Core/Portable/FindSymbols/SymbolTree/SymbolTreeInfo.cs

```diff
@@ -470,45 +455,49 @@ private static int BinarySearch(string concatenatedNames, ImmutableArray<Node> n
 
             if (_nodes[node.ParentIndex].IsRoot)
             {
-                results.AddRange(rootContainer.GetMembers(GetName(node)));
+                var members = rootContainer.GetMembers(node.Name);
+                if (!members.IsEmpty)
+                {
+                    results ??= ArrayBuilder<ISymbol>.GetInstance();
```

Comment:
this sems really odd.  why not just have the array builder created up front and passed in.  that's much more idiomatic with how we do this elsewhere.

Reply:
Mostly because it's slow, particularly for taking an object from the pool and then returning it without using it. More than 750MB of the allocations in this were the pooled objects, while only 50MB of arrays were produced by `ToImmutableAndFree`. This means one of two things is likely happening:

1. The object pool is hitting a high rate of lost objects due to parallel code returning them (this on an 8-core machine and not the 64-core machine I normally test on)
2. The objects are getting held for longer than necessary, leading to more than 128 instances in use at once

Reply:
➡️ Documented the intentional lazy initialization

Reply:
➡️ Transitioned to `TemporaryArray<T>`